Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorPendingCompositionComponent.cpp

```diff
@@ -7,18 +7,92 @@
+
+        AZ::JsonSerializationResult::Result EditorPendingCompositionComponentSerializer::Load(
+            void* outputValue,
+            [[maybe_unused]] const AZ::Uuid& outputValueTypeId,
+            const rapidjson::Value& inputValue,
+            AZ::JsonDeserializerContext& context)
+        {
+            namespace JSR = AZ::JsonSerializationResult;
+
+            AZ_Assert(
+                azrtti_typeid<EditorPendingCompositionComponent>() == outputValueTypeId,
+                "Unable to deserialize EditorPendingCompositionComponent from JSON because the provided type is %s.",
+                outputValueTypeId.ToString<AZStd::string>().c_str());
+
+            auto componentInstance = reinterpret_cast<EditorPendingCompositionComponent*>(outputValue);
+            AZ_Assert(componentInstance, "Output value for EditorPendingCompositionComponentSerializer can't be null.");
+
+            JSR::ResultCode result(JSR::Tasks::ReadField);
+            {
+                JSR::ResultCode pendingComponentsResult(JSR::Tasks::ReadField);
+
+                AZStd::unordered_map<AZStd::string, AZ::Component*> componentMap;
+
+                auto pendingComponentsIter = inputValue.FindMember("PendingComponents");
+                if (pendingComponentsIter != inputValue.MemberEnd())
+                {
+                    if (pendingComponentsIter->value.IsArray())
+                    {
+                        // If the serialized data is an array type, then convert the data to a map.
+                        AZStd::vector<AZ::Component*> componentVector;
+                        pendingComponentsResult = ContinueLoadingFromJsonObjectField(
+                            &componentVector, azrtti_typeid<decltype(componentVector)>(), inputValue, "PendingComponents", context);
+
+                        AZ::EntityUtils::ConvertComponentVectorToMap(componentVector, componentMap);
+                    }
+                    else
+                    {
+                        pendingComponentsResult = ContinueLoadingFromJsonObjectField(
+                            &componentMap, azrtti_typeid<decltype(componentMap)>(), inputValue, "PendingComponents", context);
+                    }
+
+                    static AZ::TypeId genericComponentWrapperTypeId = azrtti_typeid<GenericComponentWrapper>();
+
+                    for (auto& [componentKey, component] : componentMap)
+                    {
+                        // If the component didn't serialize (i.e. is null) or the underlying type is GenericComponentWrapper,
+                        // the template is null and the component should not be added.
+                        if (component && component->GetUnderlyingComponentType() != genericComponentWrapperTypeId)
+                        {
+                            componentInstance->m_pendingComponents.emplace(componentKey, component);
+                        }
```

Comment:
Do we have any concept of "missing" component types that are made visible in the case where someone opens a prefab that uses a component that is defined by a gem that is missing from their project - and they still see in the entity inspector that there is a component that exists on the entity but it hasn't been registered so it is disabled.  
e.g. I send you a prefab file that contains an entity with a PopcornFX component on it, you try to open it in your AutomatedTesting project which doesn't have the PopcornFX gem, does the editor show the unknown component in the entity inspector in a disabled state - or is it invisible and if the user saves it, that component is erased?

Reply:
I think it would show up a dialog saying the component cannot be loaded and it is missing in the inspector. I don't think we currently treat it as a disabled component.

<img width=400 src="https://github.com/o3de/o3de/assets/11157226/334dcb29-d967-4a1d-b4fd-8c406c132b06">